### ISSUE TITLE
[PlatformView][iOS] Touch gesture on InteractiveViewer stops working if you touch your finger first and then touch with Apple Pencil on iPad

### ISSUE THREAD

**Issue by wolru** [flutter/flutter#136244]
[Edit: internal Google bug b/341930773]

### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

If you touch with your finger first and touch with Apple Pencil second while first touch is maintained , the callback function not called.
On the contrary, if you touch with the Apple Pencil first and touch with finger later, it will work normally
Both work fine on Android

The difference is,
On Android, when the second touch is occur, the onPointerUp event of first touch is called  and the onPointerDown event of second touch is called . Not called in ipad

### Expected results

touch works expected

### Actual results

touch not working

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by huycozy** [comment]
Hi @wolru 
Please provide information in the remaining fields in the issue template. More specifically:
- A [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)
- The output of `flutter doctor -v`
- Reproduction steps corresponding to sample code

**Comment by wolru** [comment]
@huycozy Hi, 
I found a problematic situation. When you use a Webview in InteractiveViewer, touch the app pencil while touching it first with your finger will not working

https://github.com/flutter/flutter/assets/67232509/dc395a8e-3ed9-4791-a0c4-6c77da925e9d


InteractiveViewer(
  child: InAppWebView(
    initialOptions: InAppWebViewGroupOptions(
      crossPlatform: InAppWebViewOptions(
        supportZoom: false,
        javaScriptEnabled: true,
        disableHorizontalScroll: true,
        disableVerticalScroll: true,
        disableContextMenu: true,
        useShouldOverrideUrlLoading: true,
        mediaPlaybackRequiresUserGesture: false,
        transparentBackground: true,
      ),
    ),
    initialUrlRequest: URLRequest(url: Uri.parse('https://www.google.com/')),
  ),
),

**Comment by huycozy** [comment]
You are using InAppWebView which is 3rd party plugin we are not sure if the issue could be related to/caused by it. Could you retry with [webview_flutter](https://pub.dev/packages/webview_flutter) instead?

**Comment by wolru** [comment]
@huycozy The same issue occurs when using [webview_flutter](https://pub.dev/packages/webview_flutter)

**Comment by huycozy** [comment]
Please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we can verify the issue. Also, please provide the output of `flutter doctor -v` as well.

**Comment by hongmono** [comment]
> Please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we can verify the issue. Also, please provide the output of `flutter doctor -v` as well.

```
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  double scale = 1;
  double previousScale = 1;
  Offset previousOffset = Offset.zero;
  late WebViewController controller;

  @override
  void initState() {
    controller = WebViewController()..loadRequest((Uri.parse('https://www.google.com')));

    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: InteractiveViewer(
        child: IgnorePointer(
          ignoring: true,
          child: WebViewWidget(controller: controller),
        ),
      ),
    );
  }
}
```

```
[✓] Flutter (Channel stable, 3.13.8, on macOS 14.0 23A344 darwin-arm64, locale ko-KR)
    • Flutter version 3.13.8 on channel stable at /opt/homebrew/Caskroom/flutter/3.13.6/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c4930c4ac (5 days ago), 2023-10-18 10:57:55 -0500
    • Engine revision 767d8c75e8
    • Dart version 3.1.4
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/hongmono/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 75.1.4
    • Dart plugin version 232.10072.19
```

**Comment by huycozy** [comment]
Thank @hongmono for the reproduction code. However, I wonder how it can be drawn as in the demo video above. Looks like @wolru is using a draw app on the top layer?

Could you provide the reproduction step-by-step? It would be better if there is a working example (native app?)for comparison.

**Comment by hongmono** [comment]
@huycozy 

It's the same as @wolru's code: no touch event is fired after using a combination of finger gestures and apple pencil gestures on the webview, whether the webview is under the stack or as a child.

Thank you for your interest.

**Comment by huycozy** [comment]
>no touch event is fired

@hongmono Could you share the actual results of this? Do you mean the webview is not draggable? Is it caused by adding `IgnorePointer`?

Also, to identify if this is webview or platform-views issue, you can try replacing webview to pure platform-view to see if the same result occurs. Please check my simple platform-view demo [here](https://github.com/huycozy/platformviews-demo) for reference.

**Comment by wolru** [comment]
@huycozy 
Not only webview, but all touches are dead. Interactive Viewer doesn't have any touch events either

**Comment by hsd2747** [comment]
Hi @huycozy
I tried [here](https://github.com/huycozy/platformviews-demo) reference with InteractiveViewer then same issue.

If you watch the video, you'll notice that the zoom in gesture is possible at first, but after using finger and apple pencil at the same time, everything stops working.

See the video and code below
Thank you

```dart
import 'dart:io';

import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'package:flutter/services.dart';

class NativeViewIOS extends StatefulWidget {
  const NativeViewIOS({Key? key}) : super(key: key);

  @override
  State<NativeViewIOS> createState() => _NativeViewIOSState();
}

class _NativeViewIOSState extends State<NativeViewIOS> {
  Widget build(BuildContext context) {
    // This is used in the platform side to register the view.
    const String viewType = 'platformviewtype';
    // Pass parameters to the platform side.
    final Map<String, dynamic> creationParams = <String, dynamic>{};

    return Listener(
      child: InteractiveViewer(
        child: UiKitView(
          viewType: viewType,
          layoutDirection: TextDirection.ltr,
          creationParams: creationParams,
          creationParamsCodec: const StandardMessageCodec(),
        ),
      ),
    );
  }
}

```

https://github.com/flutter/flutter/assets/47204887/e54788d8-8e53-43ad-9bfd-98212445599e

**Comment by huycozy** [comment]
Thank @hsd2747, your demo is very intuitive. I can also reproduce this issue with exactly steps/gestures from your video.

One behavior I observed: When bringing app to background (Home screen) and opening it again, touch gesture works again. 

Complete sample code with pure platform-views: [touch_ios_pencil_136244](https://github.com/huycozy/platformviews-demo/tree/touch_ios_pencil_136244)

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.13.9, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.13.9 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (7 hours ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 118.0.5993.117

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.16.0-17.0.pre.39, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.16.0-17.0.pre.39 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 53a5d56805 (21 minutes ago), 2023-10-25 22:55:37 -0400
    • Engine revision 6428ed5768
    • Dart version 3.3.0 (build 3.3.0-56.0.dev)
    • DevTools version 2.28.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 118.0.5993.117

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by hellohuanlin** [comment]
@LouiseHsu and @cyanglaz Can this be related to the touch interceptor work?

**Comment by wolru** [comment]
Hi, @huycozy @cyanglaz 
When will the revised version be released?
This issue is very critical to my app. Many users are complaining abouht this problem
Thank you very much,

**Comment by sailingbin** [comment]
this issue also occurs in our product, hope it can be fix as soon as possible.

**Comment by wanjm** [comment]
@ditman  can PointerInterceptor be used as a workaround when theres is no need for PlatformView to be interactive

**Comment by ditman** [comment]
@wanjm sorry, I don't know about the behavior of iOS in this particular example.

**Comment by wanjm** [comment]
> This issue is related with ScaleGestureRecognizer. When I change the below code in gesture_detector.dart, everything is working except scailing.
> 
> ScaleGestureRecognizer(debugOwner: this, supportedDevices: supportedDevices) -> ScaleGestureRecognizer(debugOwner: this, supportedDevices: Set.of([PointerDeviceKind.stylus]))

@wolru do you mean change the above in flutter will make it work, I have try, no use;

**Comment by danyangya** [comment]
we also come across this issue , hope it can be fixed;

**Comment by wanjm** [comment]
we are terribly blocked by this bug, @cbracken  @yaakovschectman @goderbauer could you please have a look; 
or give me some guide how to avoid this;

in our scenario, we don't need webview to interact with user, is there any way to stop the event been send to webview, so to avoid the bug?

**Comment by wanjm** [comment]
> @LouiseHsu and @cyanglaz Can this be related to the touch interceptor work?

Interceptor seem doesn't help, do you have any ideas on how to avoid it; I hope there is some way to block event send to webview, and I think there should be as there is no such problem with other platformviews; 
1. I tried PointerInterceptor;
2. I tried `weoView.isUserinteractionEnabled=talse`

either seems work;

**Comment by ditman** [comment]
@wanjm I think you need to create a new issue with a sample of the code that is failing, so people can advise. It's very hard to help without reproducing the problem locally.

**Comment by wanjm** [comment]
> @wanjm I think you need to create a new issue with a sample of the code that is failing, so people can advise. It's very hard to help without reproducing the problem locally.

the above example is the same as mine;

**Comment by Tr2e** [comment]
@huycozy  I think this is a bug of PlatformView on iOS (FlutterTouchInterceptingView); I found a workaround, if your flutter_WebView does't need user interface, you can set the "userInteractionEnabled" property of webView's superview() to NO(false), your inputing operation will work good (at least in my case);
here is the code i write in FlutterWebView.m
`- (void)enableInteraction:(FlutterMethodCall*)call result:(FlutterResult)result {
    NSDictionary* arguments = [call arguments];
    int x = [arguments[@"enable"] intValue];
    if(x == 1) {
        self.view.userInteractionEnabled = YES;
        self.view.superview.userInteractionEnabled = YES;
    } else {
        self.view.userInteractionEnabled = NO;
        self.view.superview.userInteractionEnabled = NO;
    }
    result(@(x));
}`

**Comment by hellohuanlin** [comment]
> Complete sample code with pure platform-views: [touch_ios_pencil_136244](https://github.com/huycozy/platformviews-demo/tree/touch_ios_pencil_136244). The issue doesn't occur if it's not a platform-view inside InteractiveViewer.

@huycozy i downloaded this and build against the latest local engine, and got this compile error: 

> Cannot find 'FLNativeViewFactory' in scope

**Comment by huycozy** [comment]
@hellohuanlin The reason is that FLNativeView.swift and FLPlugin.swift were not included in Runner. I have solved it, please fetch again

**Comment by hellohuanlin** [comment]
@huycozy I was able to build and run this project, but I'm very confused what this project is supposed to do. It's just a button that goes to a blue screen, with "Native text from iOS" showing. I cannot experiment anything with this project. Could you clarify?

**Comment by hellohuanlin** [comment]
Update: I found a few workarounds and currently testing it to ensure it doesn't break existing touch logic. Will put up a PR soon.

**Comment by hellohuanlin** [comment]
Reopen since a customer reported that this broke the touch/gesture.

**Comment by huycozy** [comment]
It looks like this was accidentally closed by the commit linked above. cc @yaakovschectman
